setup.py

```diff
@@ -54,11 +54,11 @@
         "mike",
         "pymdown-extensions",
         # linting
-        "black",
-        "pre-commit",
-        "pylint",
-        "flake8-unused-arguments",
-        "pyright",
+        "black>=23.7.0,<24.0.0",
+        "pre-commit>=3.3.0,<4.0.0",
+        "pylint==2.17.5",
+        "flake8-unused-arguments==0.0.13",
+        "pyright==1.1.323",
```

Comment:
why not lock major versions of those?

Reply:
because even in patch versions there maybe new error detected (happened).

Reply:
for example:`pyright==1.1.316` on the branch release/2.129.0 finds errors, `pyright==1.1.300` does not.